src/redispatch.js

```diff
@@ -0,0 +1,141 @@
+/*
+The goal of this feature is to make player events work as an integrator would
+expect despite the presense of ads. For example, an integrator would expect
+an `ended` event to happen once the content is ended. If an `ended` event is sent
+as a result of an ad ending, that is a bug. The `redispatch` method should recognize
+such `ended` events and prefix them so they are sent as `adended`, and so on with
+all other player events.
+*/
+
+import videojs from 'video.js';
+
+let player;
```

Comment:
It ends up being a lot of boilerplate that I want to avoid. Of course, a class is a lot of boilerplate too.